patches/sitecustomize.py

```diff
@@ -73,3 +75,27 @@ def exec_module(self, module):
 
 if not hasattr(sys, 'frozen'):
     sys.meta_path.insert(0, GcpModuleFinder())
+
+@wrapt.when_imported('google.generativeai')
```

Comment:
Is this called only the first time this module is imported or again every time? 

To confirm, you can simply add a print statement and test in a notebook.

Reply:
It's only called the first time it seems.